### PR TITLE
Upgrade Jaeger to v1.21 and Thrift to 0.13.0 and remove gateway/etcd config

### PR DESCRIPTION
**Pull request by keitwb**
The gateway/etcd service is no longer maintained and it was blocking
the upgrade to Jaeger v1.21 due to conflicting dependencies.

We need the new verison of Jaeger to be able to upgrade thrift here and in
downstream dependencies.

### CODE REVIEW

File: protocol/signalfx/spanobfuscation/spantagremoval_test.go [hunk 1]
```diff
@@ -137,7 +138,7 @@ func BenchmarkRmTen(b *testing.B) {
 	config := make([]*TagMatchRuleConfig, 0, 10)
 	for i := 0; i < 9; i++ {
 		rule := &TagMatchRuleConfig{
-			Service:   pointer.String("some*test*service" + string(i)),
+			Service:   pointer.String("some*test*service" + strconv.Itoa(i)),
```
Comment (jgheewala) [inline protocol/signalfx/spanobfuscation/spantagremoval_test.go:141]: can we split this fix in separate commit

File: protocol/signalfx/spanobfuscation/spantagobfuscation_test.go [hunk 2]
```diff
@@ -137,7 +138,7 @@ func BenchmarkObfTen(b *testing.B) {
 	config := make([]*TagMatchRuleConfig, 0, 10)
 	for i := 0; i < 9; i++ {
 		rule := &TagMatchRuleConfig{
-			Service:   pointer.String("some*test*service" + string(i)),
+			Service:   pointer.String("some*test*service" + strconv.Itoa(i)),
```
Comment (jgheewala) [inline protocol/signalfx/spanobfuscation/spantagobfuscation_test.go:141]: can we split this fix in separate commit

**Review by jgheewala — COMMENTED** [review]
a nit else everything else lgtm. Might be worth testing this with ingest services to make sure the changes work as expected

File: protocol/signalfx/spanobfuscation/spantagremoval_test.go (hunk 1 above)
Comment (keitwb) [reply protocol/signalfx/spanobfuscation/spantagremoval_test.go]: Done in #23

**Comment by keitwb** [comment]
> Might be worth testing this with ingest services to make sure the changes work as expected

I'm going to include this update in ingest along with the Thrift 0.13 upgrade there and test it all together.
